contrib/scripts/builder.sh

```diff
@@ -5,29 +5,38 @@ set -eu
 cd "$(dirname "$0")/../.."
 
 CILIUM_BUILDER_IMAGE=$(cat images/cilium/Dockerfile | grep '^ARG CILIUM_BUILDER_IMAGE=' | cut -d '=' -f 2)
+RUN_AS_NONROOT="${RUN_AS_NONROOT:-1}"
 
 USER_OPTION=""
+USER_PATH="/root"
 
 if [ -n "${RUN_AS_NONROOT:-}" ]; then
     USER_OPTION="--user $(id -u):$(id -g)"
+    USER_PATH="$HOME/"
 fi
 
 MOUNT_GOCACHE_DIR=""
 
-if [ -n "${BUILDER_GOCACHE_DIR:-}" ]; then
-    MOUNT_GOCACHE_DIR="-v ${BUILDER_GOCACHE_DIR}:/root/.cache/go-build"
+if [ -z "${BUILDER_GOCACHE_DIR:-}" ]; then
+    MOUNT_GOCACHE_DIR="-v ${USER_PATH}/.cache/go-build/:${USER_PATH}/.cache/go-build"
```

Comment:
I think I don't have the full context in which environments this script is run, but I believe this is meant to also run on the host. If so, can we use `go env GOCACHE` here instead of `${USER_PATH}/.cache/go-build/`? I think that would be more portable. Though thinking about it, I'm not sure the go cache is OS independent itself 🤔

Reply:
Good point, `go env` would be a better solution. Fixed.

I'm also not so familiar with the different environments, but I know it's run in CI somewhere (where these `BUILDER_*` env vars will override some settings) and I know that I'd like to run it locally. Hopefully this change won't have much fallout, it will just speed up the operation for developers. If we face some challenges with this then we can revert and/or find a more elaborate solution.